### PR TITLE
docs/DeveloperGuide: update `find` command implementation details

### PR DESCRIPTION
**Pull request by lycjackie**
Let's update the `DeveloperGuide` to include details of our current
implementation and:
* include the interactions between the parser component.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -480,29 +480,78 @@ The sequence diagram below shows the interaction within the `Logic` components.
 .Find component interactions
 image::FindCommandSequenceDiagram.png[width="650"]
 
+The main implementation of this feature is split into two component. The `Tokenizer` and `BooleanExpressionParser`
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc]: the `component` needs an `s` at the back

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -480,29 +480,78 @@ The sequence diagram below shows the interaction within the `Logic` components.
...
+
+1. `Tokenizer` help to split the user provided argument into tokens which could be use by `BooleanExpressionParser`.
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc]: `help` needs an '`s` at the back

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -480,29 +480,78 @@ The sequence diagram below shows the interaction within the `Logic` components.
...
+
+The following table shows the operators currently supported by `BooleanExpressionParser`(Highest precedence
+first).
+|====
+| *Operators* | *Description*
+| `&&` | Logical AND of two predicates
+| `\|\|` | Logical OR of two predicates.
+|====
+
+Parentheses `(` and `)` are also recognized and respected, and they may be nested to arbitrary depth. This is handled
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc]: `This is handled Shunting Yard` should be rephrased to `This is handled by Shunting Yard`

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -480,29 +480,78 @@ The sequence diagram below shows the interaction within the `Logic` components.
...
+image::parserSequenceDiagram.png[width="650"]
+
 ==== Design Consideration
 
 ===== Aspect: How to parse composite predicates
 
 [NOTE]
-Currently we have use choice 1 mainly because of time constrains. When there is sufficient time, Alternative 2 would
-be more preferred because it provide our user with a more robust find command.
+Currently we have use choice 2 because this provide our users with a more flexible way of finding modules in our
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc]: maybe you were saying `Currently we are using choice 2`?

**Review by chyeo — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -480,29 +480,78 @@ The sequence diagram below shows the interaction within the `Logic` components.
...
+application.
 
-- Alternative 1 (current choice): Do an implicit boolean `OR` for every predicate.
+- Alternative 1: Do an implicit boolean `OR` for every predicate.
 
 [cols="30%,<70%"]
 |======
 |*Pros*| Very easy implementation
```
Comment (chyeo) [inline docs/DeveloperGuide.adoc]: maybe rephrased it to `Very easy to implement`?

**Review by Letm3through — APPROVED** [review]
LGTM. Approving and merging.

**Review by chyeo — APPROVED** [review]
LGTM. Thanks for updating the Developer Guide
